monitoring/procfs.go

```diff
@@ -0,0 +1,230 @@
+		if err != nil {
+			return nil, err
+		}
+		sockets = append(sockets, socket)
+	}
+	if err := lineScanner.Err(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return sockets, nil
+}
+
+func GetUDPSockets(fpath string) ([]*UDPSocket, error) {
+	var sockets []*UDPSocket
+	fp, err := os.Open(fpath)
+	defer fp.Close()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	lineScanner := bufio.NewScanner(fp)
+	lineScanner.Scan() // Drop header line
+	for lineScanner.Scan() {
+		socket, err := NewUDPSocketFromLine(lineScanner.Text())
+		if err != nil {
+			return nil, err
+		}
+		sockets = append(sockets, socket)
+	}
+	if err := lineScanner.Err(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return sockets, nil
+}
+
+func GetUnixSockets() ([]*UnixSocket, error) {
+	var sockets []*UnixSocket
+	fp, err := os.Open(ProcNetUnix)
+	defer fp.Close()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	lineScanner := bufio.NewScanner(fp)
+	lineScanner.Scan() // Drop header line
+	for lineScanner.Scan() {
+		socket, err := NewUnixSocketFromLine(lineScanner.Text())
+		if err != nil {
+			return nil, err
+		}
+		sockets = append(sockets, socket)
+	}
+	if err := lineScanner.Err(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return sockets, nil
+
+}
+
+func NewTCPSocketFromLine(line string) (*TCPSocket, error) {
+	// sl  local_address rem_address   st tx_queue rx_queue tr tm->when retrnsmt   uid  timeout inode
+	//  0: 00000000:0035 00000000:0000 0A 00000000:00000000 00:00000000 00000000     0        0 18616 1 ffff91e759d47080 100 0 0 10 0
+	// reference: https://github.com/ecki/net-tools/blob/master/netstat.c#L1070
+	var (
+		sl         int
+		localip    []byte
+		remoteip   []byte
+		tr         int
+		tmwhen     int
+		retransmit int
+		timeout    int
+		tails      string
+	)
+	tcpSocket := &TCPSocket{}
+	_, err := fmt.Sscanf(line, "%d: %X:%X %X:%X %X %X:%X %X:%X %X %d %d %d %s",
+		&sl, &localip, &tcpSocket.LocalAddress.Port, &remoteip, &tcpSocket.RemoteAddress.Port,
+		&tcpSocket.State, &tcpSocket.TXQueue, &tcpSocket.RXQueue, &tr, &tmwhen, &retransmit,
+		&tcpSocket.UID, &timeout, &tcpSocket.Inode, &tails)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	tcpSocket.LocalAddress.IP = hexToIP(localip)
+	tcpSocket.RemoteAddress.IP = hexToIP(remoteip)
+	return tcpSocket, nil
+}
+
+func NewUDPSocketFromLine(line string) (*UDPSocket, error) {
+	//    sl  local_address rem_address   st tx_queue rx_queue tr tm->when retrnsmt   uid  timeout inode ref pointer drops
+	//  2511: 00000000:14E9 00000000:0000 07 00000000:00000000 00:00000000 00000000  1000        0 1662497 2 ffff91e6a9fcbc00 0
+	var (
+		sl         int
+		localip    []byte
+		remoteip   []byte
+		tr         int
+		tmwhen     int
+		retransmit int
+		timeout    int
+	)
+	udpSocket := &UDPSocket{}
+	_, err := fmt.Sscanf(line, "%d: %X:%X %X:%X %X %X:%X %X:%X %X %d %d %d %d %X %d",
+		&sl, &localip, &udpSocket.LocalAddress.Port, &remoteip, &udpSocket.RemoteAddress.Port,
```

Comment:
Do we need all of these attributes (for instance, pointer)?
Like reading the example line on playground fails with `integer overflow` - not that it's a deal breaker but if we don't need  the attributes beyond the IP - why read and store them at all?